python/ray/serve/_private/deployment_state.py

```diff
@@ -2253,13 +2245,27 @@ def order(deadline: int):
         return to_stop, remaining
 
     def migrate_replicas_on_draining_nodes(self, draining_nodes: Dict[str, int]):
+        # Move replicas back to running if they are no longer on a draining node.
+        # If this causes the number of replicas to exceed the target state,
+        # they will be scaled down after this.
```

Comment:
Could you say how the replicas get scaled down in this comment? What component/function does the scaling down?

Reply:
Added comment explaining it